discord/ui/select.py

```diff
@@ -171,6 +194,113 @@ def max_values(self) -> int:
+
+        self._values = values[self.custom_id] = payload
+        selected_values.set(values)
+
+    def is_dispatchable(self) -> bool:
+        return True
+
+    @classmethod
+    def from_component(cls, component: SelectMenu) -> Self:
+        return cls(
+            **{k: getattr(component, k) for k in cls.__item_repr_attributes__},
+            row=None,
+        )
+
+
+class Select(BaseSelect[V]):
+    """Represents a UI select menu with a list of custom options. This is represented
+    to the user as a dropdown menu.
+
+    .. versionadded:: 2.0
+
+    Parameters
+    ------------
+    custom_id: :class:`str`
+        The ID of the select menu that gets received during an interaction.
+        If not given then one is generated for you.
+    placeholder: Optional[:class:`str`]
+        The placeholder text that is shown if nothing is selected, if any.
+    min_values: :class:`int`
+        The minimum number of items that must be chosen for this select menu.
+        Defaults to 1 and must be between 0 and 25.
+    max_values: :class:`int`
+        The maximum number of items that must be chosen for this select menu.
+        Defaults to 1 and must be between 1 and 25.
+    options: List[:class:`discord.SelectOption`]
+        A list of options that can be selected in this menu.
+    disabled: :class:`bool`
+        Whether the select is disabled or not.
+    row: Optional[:class:`int`]
+        The relative row this select menu belongs to. A Discord component can only have 5
+        rows. By default, items are arranged automatically into those 5 rows. If you'd
+        like to control the relative positioning of the row then passing an index is advised.
+        For example, row=1 will show up before row=2. Defaults to ``None``, which is automatic
+        ordering. The row number must be between 0 and 4 (i.e. zero indexed).
+    """
+
+    __item_repr_attributes__ = BaseSelect.__item_repr_attributes__ + ('options',)
+
+    def __init__(
+        self,
+        *,
+        custom_id: str = MISSING,
+        placeholder: Optional[str] = None,
+        min_values: int = 1,
+        max_values: int = 1,
+        options: List[SelectOption] = MISSING,
+        disabled: bool = False,
+        row: Optional[int] = None,
+    ) -> None:
+        super().__init__(
+            self.type,
+            custom_id=custom_id,
+            placeholder=placeholder,
+            min_values=min_values,
+            max_values=max_values,
+            disabled=disabled,
+            options=[] if options is MISSING else options,
+            row=row,
+        )
+
+    @property
+    def values(self) -> List[str]:
+        """List[:class:`str`]: A list of values that have been selected by the user."""
+        return super().values  # type: ignore
+
+    @property
+    def type(self) -> Literal[ComponentType.string_select]:
+        """:class:`.ComponentType`: The discord type of this component."""
```

Comment:
```suggestion
        """:class:`.ComponentType`: The type of this component."""
```

Same for the rest of them.